src/main/Api.ts

```diff
@@ -0,0 +1,17 @@
+import { AxiosInstance } from 'axios';
+import { User } from './interfaces/UserResultData';
+
+export class Api {
+  constructor(
+    private readonly axios: AxiosInstance
+  ) { }
+
+  public getUsersByEmail(email: string): Promise<User[]> {
+    return this.axios
+      .get(`/api/v1/users?query=email:${email}`)
```

Comment:
```suggestion
      .get('/api/v1/users', {params: { 'query=email:': email }})
```
I think using the built in params config of Axios might be safer then appending the query string to the URL

Reply:
The suggestion doesn't work so I change it to below:
".get('/api/v1/users', { params: { 'query': 'email:' + email } })"

Reply:
Sounds good to me!